expression_common_vs_other/1_expression_common_vs_other.py

```diff
@@ -0,0 +1,164 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:generic_expression] *
+#     language: python
+#     name: conda-env-generic_expression-py
+# ---
+
+# # Expression of common vs other genes
+#
+# This notebook asks if there is a technical reason for these commonly DEGs to be found as frequently differentially expressed. To answer this question, we will compare the average expression of common DEGs compared to other genes
+
+# %load_ext autoreload
+import os
+import random
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+from ponyo import utils
+from generic_expression_patterns_modules import ranking
+
+# +
+# Read in config variables
+base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
+
+config_filename = os.path.abspath(
+    os.path.join(base_dir, "configs", "config_human_general.tsv")
+)
+
+params = utils.read_config(config_filename)
+local_dir = params["local_dir"]
+dataset_name = params["dataset_name"]
+project_id = params["project_id"]
+col_to_rank_genes = params["rank_genes_by"]
+# -
+
+# Read in non (0-1) normalized expression compendium
+if dataset_name == "pseudomonas_analysis":
+    raw_compendium_filename = params["raw_compendium_filename"]
+    expression = pd.read_csv(raw_compendium_filename, sep="\t", index_col=0, header=0).T
+else:
+    mapped_compendium_filename = params["mapped_compendium_filename"]
+    expression = pd.read_csv(
+        mapped_compendium_filename, sep="\t", index_col=0, header=0
+    )
+
+expression.head()
+
+# +
+# Load gene_summary_filename
+if dataset_name == "pseudomonas_analysis":
+    gene_summary_filename = os.path.join(
+        base_dir, dataset_name, f"generic_gene_summary_{project_id}_cbrB_v_WT.tsv"
+    )
+else:
+    gene_summary_filename = os.path.join(
+        base_dir, dataset_name, f"generic_gene_summary_{project_id}.tsv"
+    )
+
+summary_gene_ranks = pd.read_csv(gene_summary_filename, sep="\t", index_col=0, header=0)
+# -
+
+summary_gene_ranks.head()
+
+# ## Get common and other genes
+
+# +
+# Get generic genes identified by Crow et. al.
+if dataset_name == "pseudomonas_analysis":
+    DE_prior_filename = os.path.join(
+        base_dir, dataset_name, params["reference_gene_filename"]
+    )
+else:
+    DE_prior_filename = params["reference_gene_filename"]
+ref_gene_col = params["reference_gene_name_col"]
+ref_rank_col = params["reference_rank_col"]
+
+figure_filename = f"gene_ranking_{col_to_rank_genes}_tmp.svg"
+
+corr, shared_ranking = ranking.compare_gene_ranking(
```

Comment:
This is exclusively array data you're plotting here, right? Interesting that you still see the group of genes in the bottom right, that went away in your last PR when you applied MR normalization to the RNA-seq data.

What kind of normalization are you doing for the array data? Do you think it could be affecting the cluster of genes in the bottom right here as well? Or is this notebook using RNA-seq data?

Reply:
This is comparing common findings from manually-curated array experiment results vs SOPHIE trained on RNA-seq data.